test/integration/query/additional.js

```diff
@@ -978,6 +978,79 @@ module.exports = function (knex) {
         });
     });
 
+    it('.timeout(ms, {cancel: true}) should cancel all parallel slow queries run in a single transaction', async function () {
+      // Only mysql/postgres query cancelling supported for now
+      if (!isMysql(knex) && !isPostgreSQL(knex)) {
+        return this.skip();
+      }
+
+      const testQueries = {
+        [drivers.PostgreSQL]: function () {
+          return knex.raw('SELECT pg_sleep(10)');
+        },
+        [drivers.MySQL]: function () {
+          return knex.raw('SELECT SLEEP(10)');
+        },
+        [drivers.MySQL2]: function () {
+          return knex.raw('SELECT SLEEP(10)');
+        },
+        [drivers.MsSQL]: function () {
+          return knex.raw("WAITFOR DELAY '00:00:10'");
+        },
+        [drivers.Oracle]: function () {
+          return knex.raw('begin dbms_lock.sleep(10); end;');
+        },
+      };
+
+      const driverName = knex.client.driverName;
+      if (!Object.prototype.hasOwnProperty.call(testQueries, driverName)) {
+        throw new Error('Missing test query for dialect: ' + driverName);
+      }
+
+      const query = testQueries[driverName]();
+
+      const getProcessesForDriver = {
```

Comment:
Maybe merge tests about timeout in describe block & put common into it?

Reply:
@maximelkin Totaly agreed. Tests for timeout need a refactor. But let's postpone it into a separate PR after merging this PR. So we can keep track what is new in this PR and what is just a refactor.